atlasdb-commons/src/test/java/com/palantir/common/concurrent/CoalescingSupplierTest.java

```diff
@@ -62,40 +60,38 @@ public class CoalescingSupplierTest {
     private final FreezableSupplier freezableDelegate = new FreezableSupplier(delegate);
     private final CoalescingSupplier<Integer> coalescing = new CoalescingSupplier<>(freezableDelegate);
 
-    private final Supplier<Integer> supplier;
-
-    public CoalescingSupplierTest(String name, Object parameter) {
-        Function<CoalescingSupplierTest, Integer> factory = (Function<CoalescingSupplierTest, Integer>) parameter;
-        supplier = () -> factory.apply(this);
+    public static List<Arguments> getParameters() {
+        return List.of(
+                Arguments.of("blocking", (Function<CoalescingSupplierTest, Integer>) test -> test.coalescing.get()),
+                Arguments.of("async", (Function<CoalescingSupplierTest, Integer>)
+                        test -> unwrap(test.coalescing.getAsync())));
     }
 
-    @Parameterized.Parameters(name = "{0}")
-    public static Object[] getParameters() {
-        return new Object[][] {
-            {"blocking", (Function<CoalescingSupplierTest, Integer>) test -> test.coalescing.get()},
-            {"async", (Function<CoalescingSupplierTest, Integer>) test -> unwrap(test.coalescing.getAsync())}
-        };
-    }
+    private Supplier<Integer> supplier;
```

Comment:
Can't all these `supplier` just be local variables? I don't see why we need it to be a private field of the class.